src/js/controller.js

```diff
@@ -150,10 +150,18 @@ export const MediaUIStates = {
   MEDIA_DURATION: {
     get: function (controller) {
       const { media } = controller;
+      const defaultDuration = controller.hasAttribute('defaultduration')
+        ? +controller.getAttribute('defaultduration')
+        : NaN;
 
       // TODO: What if duration is infinity/live? (heff)
       if (!media || !Number.isFinite(media.duration)) {
-        return NaN;
+        return defaultDuration;
+      }
+
+      // Apply defaultDuration if it's set and the media is not yet ready
+      if (!(media.readyState || Number.isNaN(defaultDuration))) {
```

Comment:
why not rely on media.duration? media.duration >= 0 and not NaN 
could backfire with custom media elements that have a limited media API

Reply:
Interesting thought! I think `readyState` better represents the actual conditions, but only relying on `duration` sounds like it would be good simply because it reduces the barrier of entry for custom slotted media elements' API. _**technically**_ 0 duration media is valid, but I think we can ignore that wild scenario unless/until it comes up. Sold.

Reply:
right `0` would be wild :) was thinking current time